src/vs/workbench/contrib/terminal/electron-sandbox/localTerminalBackend.ts

```diff
@@ -101,44 +100,42 @@ class LocalTerminalBackend extends BaseTerminalBackend implements ITerminalBacke
 			// used for pty host management messages, it would make sense in the future to use a
 			// separate interface/service for this one.
 			const client = new MessagePortClient(port, `window:${this._environmentService.window.id}`);
-			const proxy = ProxyChannel.toService<IPtyService>(client.getChannel(TerminalIpcChannels.PtyHostWindow));
-			this._proxy = proxy;
-			this._proxyReady.complete();
+			this._clientEventually.complete(client);
 
 			// Attach process listeners
-			proxy.onProcessData(e => this._ptys.get(e.id)?.handleData(e.event));
-			proxy.onDidChangeProperty(e => this._ptys.get(e.id)?.handleDidChangeProperty(e.property));
-			proxy.onProcessExit(e => {
+			this._proxy.onProcessData(e => this._ptys.get(e.id)?.handleData(e.event));
+			this._proxy.onDidChangeProperty(e => this._ptys.get(e.id)?.handleDidChangeProperty(e.property));
+			this._proxy.onProcessExit(e => {
 				const pty = this._ptys.get(e.id);
 				if (pty) {
 					pty.handleExit(e.event);
 					this._ptys.delete(e.id);
 				}
 			});
-			proxy.onProcessReady(e => this._ptys.get(e.id)?.handleReady(e.event));
-			proxy.onProcessReplay(e => this._ptys.get(e.id)?.handleReplay(e.event));
-			proxy.onProcessOrphanQuestion(e => this._ptys.get(e.id)?.handleOrphanQuestion());
-			proxy.onDidRequestDetach(e => this._onDidRequestDetach.fire(e));
+			this._proxy.onProcessReady(e => this._ptys.get(e.id)?.handleReady(e.event));
+			this._proxy.onProcessReplay(e => this._ptys.get(e.id)?.handleReplay(e.event));
+			this._proxy.onProcessOrphanQuestion(e => this._ptys.get(e.id)?.handleOrphanQuestion());
+			this._proxy.onDidRequestDetach(e => this._onDidRequestDetach.fire(e));
```

Comment:
are these disposables? Do they need to be registered as such?

Reply:
The backend owns the proxy so we could expose that, but we don't currently as the object never actually gets disposed.